homeassistant/components/devolo_home_control/__init__.py

```diff
@@ -29,21 +29,9 @@ async def async_setup_entry(
     """Set up the devolo account from a config entry."""
     mydevolo = configure_mydevolo(entry.data)
 
-    credentials_valid = await hass.async_add_executor_job(mydevolo.credentials_valid)
-
-    if not credentials_valid:
-        raise ConfigEntryAuthFailed(
-            translation_domain=DOMAIN,
-            translation_key="invalid_auth",
-        )
-
-    if await hass.async_add_executor_job(mydevolo.maintenance):
-        raise ConfigEntryNotReady(
-            translation_domain=DOMAIN,
-            translation_key="maintenance",
-        )
-
-    gateway_ids = await hass.async_add_executor_job(mydevolo.get_gateway_ids)
+    gateway_ids = await hass.async_add_executor_job(
+        check_mydevolo_and_get_gateway_ids, mydevolo
+    )
 
     if entry.unique_id and GATEWAY_SERIAL_PATTERN.match(entry.unique_id):
         uuid = await hass.async_add_executor_job(mydevolo.uuid)
```

Comment:
can we also add this one?

Reply:
This conversion of the unique ID was introduced in https://github.com/home-assistant/core/pull/43005 5 years ago. I would assume, that it's not happening anymore because all setups were migrated. Mind if I remove it in a seperate PR instead?

Reply:
Sounds good!